test/connectors/ArrivalMovementConnectorSpec.scala

```diff
@@ -93,6 +95,57 @@ class ArrivalMovementConnectorSpec extends SpecBase with WireMockServerHandler w
         )
       }
     }
+
+    "getPDF" - {
+      "must return status Ok and PDF" in {
+
+        val arrivalId = ArrivalId(0)
+
+        server.stubFor(
+          get(urlEqualTo(s"/$startUrl/movements/arrivals/${arrivalId.index}/unloading-permission"))
+            .willReturn(
+              aResponse()
+                .withStatus(200)
+            )
+        )
+
+        val result: Future[Option[Array[Byte]]] = connector.getPDF(arrivalId)
+
+        result.futureValue.value mustBe an[Array[Byte]]
+      }
+
+      "must return None for an unauthorized response" in {
+
+        val genErrorResponse             = Gen.oneOf(300, 500).sample.value
+        val arrivalId                    = ArrivalId(0)
+        val hcWithoutAuth: HeaderCarrier = HeaderCarrier()
+
+        val result: Future[Option[Array[Byte]]] = connector.getPDF(arrivalId)(hcWithoutAuth)
+
+        result.futureValue mustBe None
+      }
+
+      "must return status error response" in {
+
+        val genErrorResponse = Gen.oneOf(300, 500).sample.value
+        val arrivalId        = ArrivalId(0)
+
+        server.stubFor(
+          get(urlEqualTo(s"/$startUrl/movements/arrivals/${arrivalId.index}/unloading-permission"))
+            .willReturn(
+              aResponse()
+                .withStatus(genErrorResponse)
+            )
+        )
+
+        val result: Future[Option[Array[Byte]]] = connector.getPDF(arrivalId)
+
+        whenReady(result.failed) {
+          response =>
+            response mustBe an[NoSuchElementException]
+        }
+      }
```

Comment:
Can we handle this more gracefully rather than it bubbling up and showing the user technical difficulties? We would need to get a design to work on an error page that tells the user that there is a temporary problem and to try again later (if it's a 502 when the pdf service fails or a 500 when the backend fails)

Reply:
Hey @amaalali, I have included a quick walk around for this by redirecting to the Technical Difficulties controller. I will have a dicussion with design and potentially raise a seperate task to complete this. Would it be possible to merge this logic in now given that only the redirect location will change in the future?